pkg/clusteragent/admission/mutate/common/common.go

```diff
@@ -157,14 +153,24 @@ func InjectVolume(pod *corev1.Pod, volume corev1.Volume, volumeMount corev1.Volu
 			continue
 		}
 		pod.Spec.InitContainers[i].VolumeMounts = append(pod.Spec.InitContainers[i].VolumeMounts, volumeMount)
-		shouldInject = true
+		injectedVolumeMount = true
 	}
 
-	if shouldInject {
+	volumeAlreadyExists := false
+	for _, vol := range pod.Spec.Volumes {
+		if vol.Name == volume.Name {
+			log.Debugf("Pod %q: Volume %q already exists. Not adding again", podStr, vol.Name)
+			volumeAlreadyExists = true
+		}
+	}
+
+	shouldInjectVolume := !volumeAlreadyExists && injectedVolumeMount
```

Comment:
Why is `injectedVolumeMount` needed here?
If it’s set to false, that means the expected volume mounts were already there.
But shouldn’t the volume be injected anyway if it doesn’t already exist?